Samples/Desktop/D3D12Raytracing/src/D3D12RaytracingSakuraScene/D3D12RaytracingSakuraScene.cpp

```diff
@@ -0,0 +1,1854 @@
+        missShaderIdentifier = stateObjectProperties->GetShaderIdentifier(c_missShaderName);
+        floorHitGroupShaderIdentifier = stateObjectProperties->GetShaderIdentifier(c_floorHitGroupName);
+        trunkHitGroupShaderIdentifier = stateObjectProperties->GetShaderIdentifier(c_trunkHitGroupName);
+        leavesHitGroupShaderIdentifier = stateObjectProperties->GetShaderIdentifier(c_leavesHitGroupName);
+        bushHitGroupShaderIdentifier = stateObjectProperties->GetShaderIdentifier(c_bushHitGroupName);
+        tcubeHitGroupShaderIdentifier = stateObjectProperties->GetShaderIdentifier(c_transparentCubeHitGroupName);
+    };
+
+    // Get shader identifiers.
+    UINT shaderIdentifierSize;
+    {
+        ComPtr<ID3D12StateObjectProperties> stateObjectProperties;
+        ThrowIfFailed(m_dxrStateObject.As(&stateObjectProperties));
+        GetShaderIdentifiers(stateObjectProperties.Get());
+        shaderIdentifierSize = D3D12_SHADER_IDENTIFIER_SIZE_IN_BYTES;
+    }
+
+    // Ray gen shader table
+    {
+        struct RootArguments 
+        {
+            ObjectConstantBuffer cb;
+        } rootArguments;
+        rootArguments.cb = m_objectCB;
+
+        UINT numShaderRecords = 1;
+        UINT shaderRecordSize = shaderIdentifierSize + sizeof(rootArguments);
+        ShaderTable rayGenShaderTable(device, numShaderRecords, shaderRecordSize, L"RayGenShaderTable");
+        rayGenShaderTable.push_back(ShaderRecord(rayGenShaderIdentifier, shaderIdentifierSize, &rootArguments, sizeof(rootArguments)));
+        m_rayGenShaderTable = rayGenShaderTable.GetResource();
+    }
+
+    // Miss shader table
+    {
+        UINT numShaderRecords = 1;
+        UINT shaderRecordSize = shaderIdentifierSize;
+        ShaderTable missShaderTable(device, numShaderRecords, shaderRecordSize, L"MissShaderTable");
+        missShaderTable.push_back(ShaderRecord(missShaderIdentifier, shaderIdentifierSize));
+        m_missShaderTable = missShaderTable.GetResource();
+    }
+
+    // Hit group shader table
+    {
+        struct RootArguments 
+        {
+            ObjectConstantBuffer cb;
+        };
+
+		// Total number of shader records in the hit group shader table
+        UINT numShaderRecords = 4285;
+        UINT shaderRecordSize = shaderIdentifierSize + sizeof(RootArguments);
+        ShaderTable hitGroupShaderTable(device, numShaderRecords, shaderRecordSize, L"HitGroupShaderTable");
+
+        //// Larger cube shader records as the ground
+        for (int i = 0; i < 441; ++i) 
+        {
+            RootArguments argument;
+            argument.cb = m_cubeCB;
+            argument.cb.albedo = XMFLOAT4(1.0f, 1.0f, 1.0f, 1.0f);
+            argument.cb.materialID = 0;
+            hitGroupShaderTable.push_back(ShaderRecord(floorHitGroupShaderIdentifier, shaderIdentifierSize, &argument, sizeof(argument)));
+        }
+
+		// Transparant cube shader records randomly placed around the scene
+        for (int i = 0; i < 961; ++i) 
```

Comment:
hardcoded constants should be #defines or const UINT to be more maintainable/readable